ca_driver/src/create_driver.cpp

```diff
@@ -222,6 +224,16 @@ void CreateDriver::undockCallback(const std_msgs::EmptyConstPtr& msg)
   robot_->setMode(create::MODE_FULL);
 }
 
+void CreateDriver::defineSongCallback(const ca_msgs::DefineSongConstPtr& msg)
+{
+  robot_->defineSong(msg->song, msg->length, &(msg->notes.front()), &(msg->durations.front()));
```

Comment:
This function returns a bool. At the very least we can check the return type and output an error message on failure. Something like `ROS_ERROR_STREAM("[CREATE] Failed to define song " << msg->song << " of length " << msg->length);`